examples/seismic/model.py

```diff
@@ -0,0 +1,126 @@
+import numpy as np
+
+from devito import DenseData
+
+
+__all__ = ['Model']
+
+
+def damp_boundary(damp, nbpml, spacing):
+    """Initialise damping field with an absorbing PML layer.
+
+    :param damp: Array data defining the damping field
+    :param nbpml: Number of points in the damping layer
+    :param spacing: Grid spacing coefficent
+    """
+    dampcoeff = 1.5 * np.log(1.0 / 0.001) / (40 * spacing)
+    ndim = len(damp.shape)
+    for i in range(nbpml):
+        pos = np.abs((nbpml - i + 1) / float(nbpml))
+        val = dampcoeff * (pos - np.sin(2*np.pi*pos)/(2*np.pi))
+        if ndim == 2:
+            damp[i, :] += val
+            damp[-(i + 1), :] += val
+            damp[:, i] += val
+            damp[:, -(i + 1)] += val
+        else:
+            damp[i, :, :] += val
+            damp[-(i + 1), :, :] += val
+            damp[:, i, :] += val
+            damp[:, -(i + 1), :] += val
+            damp[:, :, i] += val
+            damp[:, :, -(i + 1)] += val
+
+
```

Comment:
As we do have an explicit expression for this, could we change it to a separate loop instead and avoid a useless computation on zeros? Not sure if it should be part of this PR but worth mentioning.